python/lsst/sims/maf/utils/obs2sqlite.py

```diff
@@ -0,0 +1,182 @@
+import numpy as np
+import pandas as pd
+from lsst.sims.skybrightness import stupidFast_RaDec2AltAz, SkyModel
+import lsst.sims.skybrightness_pre as sb
+from lsst.sims.utils import raDec2Hpid, m5_flat_sed, Site
+import healpy as hp
+import sqlite3
+import ephem
+import sys
+
+__all__ = ['obs2sqlite']
+
+
+class mjd2night(object):
+    def __init__(self, mjd_start=59580.035):
+        self.site = Site(name='LSST')
+        self.obs = ephem.Observer()
+        self.obs.lat = self.site.latitude_rad
+        self.obs.lon = self.site.longitude_rad
+        self.obs.elevation = self.site.height
+        self.mjd = mjd_start
+        self.sun = ephem.Sun()
+        self.generate_sunsets()
+
+    def generate_sunsets(self, nyears=13, day_pad=50):
+        """
+        Generate the sunrise times for LSST so we can label nights by MJD
+        """
+
+        # Set observatory horizon to zero
+        doff = ephem.Date(0)-ephem.Date('1858/11/17')
+
+        self.obs.horizon = 0.
+
+        # Swipe dates to match sims_skybrightness_pre365
+        mjd_start = self.mjd
+        mjd_end = np.arange(59560, 59560+365.25*nyears+day_pad+366, 366).max()
+        step = 0.25
+        mjds = np.arange(mjd_start, mjd_end+step, step)
+        setting = mjds*0.
+
+        # Stupid Dublin Julian Date
+        djds = mjds - doff
+        sun = ephem.Sun()
+
+        for i, (mjd, djd) in enumerate(zip(mjds, djds)):
+            sun.compute(djd)
+            setting[i] = self.obs.previous_setting(sun, start=djd, use_center=True)
+        setting = setting + doff
+
+        # zomg, round off crazy floating point precision issues
+        setting_rough = np.round(setting*100.)
+        u, indx = np.unique(setting_rough, return_index=True)
+        self.setting_sun_mjds = setting[indx]
+        left = np.searchsorted(self.setting_sun_mjds, mjd_start)
+        self.setting_sun_mjds = self.setting_sun_mjds[left:]
```

Comment:
So while this is very cool and definitely the accurate way to go .. what's wrong with just figuring out where noon is via the timezone, and using that to split times before noon / after noon into one day / next day?